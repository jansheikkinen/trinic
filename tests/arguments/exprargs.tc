// Expression Arguments

function main(void) -> void do
  let x, y, z: int8 = undefined;
  let a, b, c: int8 = 1 + 2, 3, 4;
end
