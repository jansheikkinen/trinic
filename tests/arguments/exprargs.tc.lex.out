
### LEXER ###
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:          LET,     (null), (004, 005)
009:   IDENTIFIER,          x, (004, 007)
010:            ,,     (null), (004, 008)
011:   IDENTIFIER,          y, (004, 010)
012:            ,,     (null), (004, 011)
013:   IDENTIFIER,          z, (004, 013)
014:            :,     (null), (004, 014)
015:         INT8,     (null), (004, 019)
016:            =,     (null), (004, 021)
017:    UNDEFINED,     (null), (004, 031)
018:            ;,     (null), (004, 032)
019:          LET,     (null), (005, 005)
020:   IDENTIFIER,          a, (005, 007)
021:            ,,     (null), (005, 008)
022:   IDENTIFIER,          b, (005, 010)
023:            ,,     (null), (005, 011)
024:   IDENTIFIER,          c, (005, 013)
025:            :,     (null), (005, 014)
026:         INT8,     (null), (005, 019)
027:            =,     (null), (005, 021)
028:      INTEGER,          1, (005, 024)
029:            +,     (null), (005, 025)
030:      INTEGER,          2, (005, 028)
031:            ,,     (null), (005, 028)
032:      INTEGER,          3, (005, 031)
033:            ,,     (null), (005, 031)
034:      INTEGER,          4, (005, 034)
035:            ;,     (null), (005, 034)
036:          END,     (null), (006, 003)
